server/aws-lsp-partiql/src/server/completion-hint/parser-completion.ts

```diff
@@ -0,0 +1,153 @@
+import { CommonTokenStream, CharStream, ParseTree, TerminalNode } from 'antlr4ng'
+import { PartiQLParser } from '../../antlr-generated/PartiQLParser'
+import { PartiQLTokens } from '../../antlr-generated/PartiQLTokens'
+import { CompletionList, CompletionItem } from '@aws/language-server-runtimes/server-interface'
+import { CodeCompletionCore } from 'antlr4-c3'
+
+const ignoredTokens = [
+    PartiQLParser.CARET,
+    PartiQLParser.COMMA,
+    PartiQLParser.PLUS,
+    PartiQLParser.MINUS,
+    PartiQLParser.SLASH_FORWARD,
+    PartiQLParser.PERCENT,
+    PartiQLParser.AT_SIGN,
+    PartiQLParser.TILDE,
+    PartiQLParser.ASTERISK,
+    PartiQLParser.VERTBAR,
+    PartiQLParser.AMPERSAND,
+    PartiQLParser.BANG,
+    PartiQLParser.LT_EQ,
+    PartiQLParser.GT_EQ,
+    PartiQLParser.EQ,
+    PartiQLParser.NEQ,
+    PartiQLParser.CONCAT,
+    PartiQLParser.ANGLE_LEFT,
+    PartiQLParser.ANGLE_RIGHT,
+    PartiQLParser.ANGLE_DOUBLE_LEFT,
+    PartiQLParser.ANGLE_DOUBLE_RIGHT,
+    PartiQLParser.BRACKET_LEFT,
+    PartiQLParser.BRACKET_RIGHT,
+    PartiQLParser.BRACE_LEFT,
+    PartiQLParser.BRACE_RIGHT,
+    PartiQLParser.PAREN_LEFT,
+    PartiQLParser.PAREN_RIGHT,
+    PartiQLParser.COLON,
+    PartiQLParser.COLON_SEMI,
+    PartiQLParser.QUESTION_MARK,
+    PartiQLParser.PERIOD,
+]
+
+function initializePartiQLParser(content: string) {
+    const inputStream = CharStream.fromString(content)
+    const lexer = new PartiQLTokens(inputStream)
+    const tokenStream = new CommonTokenStream(lexer)
+    const parser = new PartiQLParser(tokenStream)
+    // Override error listener as we only want completion behavior.
+    parser.removeErrorListeners()
+    return parser
+}
+
+function symbolAtCaretPosition(parseTree: TerminalNode, caretPosition: { line: number; character: number }) {
+    const start = parseTree.symbol.column
+    const stop = start + (parseTree.symbol.text?.length ?? 0)
+
+    return (
+        parseTree.symbol.line == caretPosition.line + 1 &&
```

Comment:
I think this `+ 1` here is to account for ANTLR lines being indexed from 1? It might make it a bit clearer to do the translation from LSP coords -> ANTLR coords a bit earlier, maybe as part of the `getPosition` method?

Reply:
It is a smart way to clean up the code. Appreciate the suggestion!